src/smartpeak/source/core/RawDataProcessor.cpp

```diff
@@ -2413,4 +2419,246 @@ namespace SmartPeak
 
     LOGD << "END CalculateMDVAccuracies";
   }
+
+  ParameterSet PickMS2Features::getParameterSchema() const
+  {
+    OpenMS::MassTraceDetection mass_trace_detection;
+    OpenMS::ElutionPeakDetection elution_peak_detection;
+    OpenMS::FeatureFindingMetabo feature_finding_metabo;
+    ParameterSet parameters({ mass_trace_detection, elution_peak_detection, feature_finding_metabo });
+    std::map<std::string, std::vector<std::map<std::string, std::string>>> param_struct({
+    {"PickMS2Features", {
+    {
+      {"name", "enable_elution"},
+      {"type", "bool"},
+      {"value", "true"},
+      {"description", "Enable elution peak detection"},
+    }
+    }} });
+    ParameterSet pick_ms2_feature_params(param_struct);
+    parameters.merge(pick_ms2_feature_params);
+    return parameters;
+  }
+
+  void PickMS2Features::process(
+    RawDataHandler& rawDataHandler_IO,
+    const ParameterSet& params_I,
+    const Filenames& filenames
+  ) const
+  {
+    LOGD << "START PickMS2Features";
+
+    if (params_I.count("PickMS2Features") && params_I.at("PickMS2Features").empty()) {
+      LOGE << "No parameters passed to PickMS2Features.";
+      LOGD << "END PickMS2Features";
+      return;
```

Comment:
I know that this approach is consistent with other process methods, however it seems to be hard for integration. No throw no return, no error flag, it seems to be imposible to programically detect error from caller point of view.

Reply:
yes I agree.